pkg/middlewares/iam_policy_attachment_expander.go

```diff
@@ -53,14 +53,42 @@ func (m IamPolicyAttachmentSanitizer) Execute(remoteResources, resourcesFromStat
 	return nil
 }
 
-func (m IamPolicyAttachmentSanitizer) sanitize(policyAttachment *resource.AbstractResource) []resource.Resource {
+func (m IamPolicyAttachmentExpander) expand(policyAttachment *resource.AbstractResource) []resource.Resource {
+
+	arn, _ := policyAttachment.Attrs.Get("policy_arn")
+	user, exist := policyAttachment.Attrs.Get("user")
+	if exist {
+		user := user.(string)
+		newAttachment := m.resourceFactory.CreateAbstractResource(
+			resourceaws.AwsIamPolicyAttachmentResourceType,
+			fmt.Sprintf("%s-%s", user, arn),
+			map[string]interface{}{
+				"users": []string{user},
+			},
+		)
+		return []resource.Resource{newAttachment}
+	}
+
+	role, exist := policyAttachment.Attrs.Get("role")
+	if exist {
+		role := role.(string)
+		newAttachment := m.resourceFactory.CreateAbstractResource(
+			resourceaws.AwsIamPolicyAttachmentResourceType,
+			fmt.Sprintf("%s-%s", role, arn),
+			map[string]interface{}{
+				"roles": []string{role},
+			},
+		)
+		return []resource.Resource{newAttachment}
+	}
 
 	var newResources []resource.Resource
 
 	users := (*policyAttachment.Attrs)["users"]
 	if users != nil {
 		// we create one attachment per user
-		for _, user := range users.([]string) {
+		for _, user := range users.([]interface{}) {
+			user := user.(string)
```

Comment:
Why not casting users instead ? `for _, user := range users.([]string) {`

Reply:
Because we have a `[]interface{}` not `[]string{}`

![image](https://user-images.githubusercontent.com/6154987/118463408-02cffc00-b700-11eb-8cd6-4193f00d4b16.png)